### PR TITLE
Removed Reputation::slash() implementation

### PR DESCRIPTION
**Pull request by jayeshbairagi**
- Removed Reputation::slash() implementation from contracts
- Removed Reputation::slash() test cases

Fixes: #172

### CODE REVIEW

File: contracts/reputation/Reputation.sol [hunk 1]
```diff
@@ -458,37 +458,6 @@ contract Reputation is MasterCopyNonUpgradable, ConsensusModule {
...
-     *          - a validator has not withdrawn
-     *
-     * @param _validator A validator address to slash.
-     */
-    function slash(address _validator)
-        external
-        onlyConsensus
-        hasStaked(_validator)
-        hasNotWithdrawn(_validator)
-        hasNotSlashed(_validator)
```
Comment (abhayks1) [inline contracts/reputation/Reputation.sol:476]: hasNotSlashed modifier is not used anymore. We can removed this modifier.

**Review by abhayks1 — DISMISSED** [review]
Nice work 👍 
Please see inline comment.

File: contracts/reputation/Reputation.sol (hunk 1 above)
Comment (jayeshbairagi) [reply contracts/reputation/Reputation.sol:476]: Done

Comment (benjaminbollen) [reply contracts/reputation/Reputation.sol:476]: resolved

File: test/reputation/cashout_earnings.js [hunk 2]
```diff
@@ -179,6 +164,6 @@ contract('Reputation::cashoutEarnings', (accounts) => {
       amount,
       { from: validator.address },
     ),
-    'Validator has withdrawn.');
+      'Validator has withdrawn.');
```
Comment (benjaminbollen) [inline test/reputation/cashout_earnings.js:167]: indentation seems off

Comment (benjaminbollen) [reply test/reputation/cashout_earnings.js:167]: but done by standardJS (per @jayeshbairagi ) so ok for me

**Review by benjaminbollen — APPROVED** [review]
LGTM

**Pull request merged**